tests/unit/subscriptions/test-class-wc-payments-product-service.php

```diff
@@ -286,13 +286,32 @@ public function test_get_wcpay_product_id_option() {
 	}
 
 	/**
-	 * Tests for WC_Payments_Product_Service::get_wcpay_price_id_option()
+	 * Tests for WC_Payments_Product_Service::get_wcpay_product_id_for_item()
 	 */
-	public function test_get_wcpay_price_id_option() {
-		$this->assertSame( '_wcpay_product_price_id_live', WC_Payments_Product_Service::get_wcpay_price_id_option() );
+	public function test_get_wcpay_product_id_for_item() {
 
-		// set to testmode.
-		WC_Payments::get_gateway()->update_option( 'test_mode', 'yes' );
-		$this->assertSame( '_wcpay_product_price_id_test', WC_Payments_Product_Service::get_wcpay_price_id_option() );
+		$this->mock_api_client->expects( $this->once() )
+		->method( 'create_product' )
+		->willReturn(
+			[
+				'wcpay_product_id' => 'product_id_test123',
+				'wcpay_price_id'   => 'price_test123',
+			]
+		);
+
+		// If type is 'Test Tax *&^ name', the result should be _wcpay_product_id_live_test_tax__name.
+		$test_type = 'Test Tax *&^ name';
+		$this->product_service->get_wcpay_product_id_for_item( $test_type );
+
+		$this->assertFalse( get_option( '_wcpay_product_id_live_Test Tax *&^ name' ) );
+		$this->assertSame( 'product_id_test123', get_option( '_wcpay_product_id_live_test_tax__name' ) );
+	}
+
+	/**
+	 * Tests for WC_Payments_Product_Service::sanitize_option_key()
+	 */
+	public function test_sanitize_option_key() {
```

Comment:
IMO we can drop this test case, as `test_get_wcpay_product_id_for_item` covers the logic.

Reply:
@kalessil - I have removed the redundant test case, just now, here https://github.com/Automattic/woocommerce-payments/pull/4496/commits/e792fc0b78e0f64068513d65fe0e3ef4c8bc5d18

Let me know if this looks good to merge. Thanks :)